cub/test/catch2_test_device_scan.cuh

```diff
@@ -56,3 +90,120 @@ void compute_inclusive_scan_reference(InputIt first,
     *result++ = acc;
```

Comment:
important: this needs a cast to `OutputIt` value type

Reply:
suggestion: we might want accumulator type in this function to be computed as `cub::detail::accumulator_t`

Reply:
important: same notes apply to `compute_exclusive_scan_reference`

Reply:
Thanks! Added `cast`s and explicit `accumulator_t`. Feel free to give it a look, if you like 🙂